metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistration.java

```diff
@@ -337,7 +341,7 @@ public boolean equals(Object o) {
             other.fenced == fenced &&
             other.inControlledShutdown == inControlledShutdown &&
             other.isMigratingZkBroker == isMigratingZkBroker &&
-            other.directories.equals(directories);
+            other.sortedDirectories.equals(sortedDirectories);
```

Comment:
This is another reason to go with sorted -- `list.equals` won't do the right thing otherwise.